src/components/academy/grading/GradingWorkspace.tsx

```diff
@@ -180,33 +194,38 @@ class GradingWorkspace extends React.Component<GradingWorkspaceProps> {
       this.props.storedQuestionId !== questionId
     ) {
       const question = this.props.grading[questionId].question as IQuestion;
+      const autogradingResults =
```

Comment:
Ternary operators are to be only used for really obvious cases. It hinders readability and its bad practice in general (esp stacking them like how you have done here!)

Lets make things explicit:
```
    let autogradingResults : AutogradingResult[] | undefined;
    let editorValue: string | undefined = ""; 
    let editorPrepend: string | null = "";
    let editorPostpend: string | null = "";
    let editorTestcases: ITestcase[] = [];

    if (question.type === QuestionTypes.programming) {
      autogradingResults = (question as IAnsweredQuestion).autogradingResults;

      editorValue = ((question as IAnsweredQuestion).answer as string);
      if (!editorValue) {
          editorValue = (question as IAnsweredQuestion).solutionTemplate;
      }

      if ((question as IAnsweredQuestion).prepend) {
        editorPrepend = (question as IAnsweredQuestion).prepend;
      }

      if ((question as IAnsweredQuestion).postpend) {
        editorPostpend = (question as IAnsweredQuestion).postpend;
      }

      if ((question as IAnsweredQuestion).testcases) {
        editorTestcases = (question as IAnsweredQuestion).testcases;
      }
    }
```

Reply:
Done as requested